libp2p/crypto/p2pSigner.go

```diff
@@ -1,64 +1,74 @@
 package crypto
 
 import (
-	"fmt"
+	"crypto/sha256"
 
 	"github.com/ElrondNetwork/elrond-go-core/core"
 	"github.com/ElrondNetwork/elrond-go-core/core/check"
 	crypto "github.com/ElrondNetwork/elrond-go-crypto"
-	libp2pCrypto "github.com/libp2p/go-libp2p-core/crypto"
-	"github.com/libp2p/go-libp2p-core/peer"
 )
 
-type p2pSigner struct {
-	privateKey libp2pCrypto.PrivKey
+// ArgsP2pSignerWrapper defines the arguments needed to create a p2p signer wrapper
+type ArgsP2pSignerWrapper struct {
+	PrivateKey crypto.PrivateKey
+	Signer     crypto.SingleSigner
+	KeyGen     crypto.KeyGenerator
 }
 
-// NewP2PSigner creates a new p2pSigner instance
-func NewP2PSigner(privateKey libp2pCrypto.PrivKey) (*p2pSigner, error) {
-	if check.IfNilReflect(privateKey) {
+type p2pSignerWrapper struct {
+	privateKey crypto.PrivateKey
+	signer     crypto.SingleSigner
+	keyGen     crypto.KeyGenerator
+}
+
+// NewP2PSignerWrapper creates a new p2pSigner instance
+func NewP2PSignerWrapper(args ArgsP2pSignerWrapper) (*p2pSignerWrapper, error) {
+	if check.IfNilReflect(args.PrivateKey) {
```

Comment:
perhaps extract to a `checkArgs` method?

also, all checks can be used as `check.IfNil` as all interfaces implement `IsInterfaceNil` method

Reply:
done